debian/rules

```diff
@@ -4,6 +4,7 @@ export DH_VERBOSE=1
 export DESTDIR=$(CURDIR)/debian/tmp
 
 export DEB_HOST_ARCH      ?= $(shell dpkg-architecture -qDEB_HOST_ARCH)
+extraopts += -DCMAKE_C_COMPILER=gcc-11 -DCMAKE_CXX_COMPILER=g++-11
```

Comment:
i don't like hard-coding the compiler version like this, but i'm not sure what else to do if we can't rely on the global `update-alternatives`

`ceph.spec.in` is effectively doing the same with `. /opt/rh/gcc-toolset-11/enable`

Reply:
since `debian/rules` is practically a GNU Makefile, we could set a default makefile variable for the used toolset. better flexibility this way,  but, yeah, it's still hard-coded.

regarding, `ceph.spec.in`, IIRC, the gcc-toolset comes with a rpm macro, which helps to setup the environment variables, more abstraction less hardwiring this way, but again, the rpm macro is the new "magic number".

Reply:
i feel like `update-alternatives` really is the right way to handle this - in CI scripts instead of the package/build system itself

the problem remains with `update-alternatives` and stable branches, but we could potentially fix that by backporting a change to `install-deps.sh` that restores the default back to gcc-8 - then master would be free to change it as needed

think that could work?

Reply:
yeah. that'd work. just couple downsides:

- less self-contained.
- `update-alternatives` only works for debian and its derivatives. 
- probably not very friendly to those who want to build packages by themselves -- they would need to prepare the environment first.

regarding the cleanup step, we could do the cleanup after building packages using a bash trap or something.